islandora.module

```diff
@@ -181,7 +181,8 @@ function islandora_file_insert(FileInterface $file) {
  */
 function islandora_file_update(FileInterface $file) {
   // Exit early if unchanged.
-  if ($file->filehash != NULL && $file->original->filehash != NULL && $file->filehash['sha1'] == $file->original->filehash['sha1']) {
+  if ($file->hasField('sha1') && $file->original->hasField('sha1')
```

Comment:
I put a breakpoint here and it did indeed exit the function early at this point, but this did not prevent the derivatives from being regenerated.

Reply:
Ah, but it does prevent it from generating each derivate twice. So, this is working as intended, my expectation of the original file being replaced with the same file not triggering any derivative generation is out of scope for this pull (and was never a feature as far as I can tell).

Reply:
If the arbitrary syntax makes any sense, this was what i traced (with a breakpoint at every update/insert hook, and a breakpoint at EmitEvent to tell me what was happening). The derivatives avoided above would have run later, in hook_media_insert (or update):

```
[During media creation]

hook_file_insert
-> Update -> indexing-fcrepo
-> Update -> indexing-triplestore

[Media save]

hook_file_update
+++ executing derivative reactions +++
-> Generate derivative -> houdini - service file from original
-> Generate derivative -> houdini - thumbnail from original
-> generate derivative -> fits - fits

hook_media_insert
+++ Execute Media Reaction +++
-> Update -> indexing-fcrepo
-> Update -> indexing-triplestore

+++ Execute Derivative Reactions +++
-> Generate derivative -> houdini - service file from original
-> Generate derivative -> houdini - thumbnail from original
-> generate derivative -> fits - fits

+++ _islandora_fire_media_file_derivative_reaction
(we had no multifile media derivatives configured, and none ran.)
```